### PR TITLE
Refactor plugins to own folder

### PR DESCRIPTION
**Pull request by janetchen1**
Put all plugins in "plugin" folder with index.js that exports one function to handle all plugin operations. Implemented structure for whitelisting channels, though no channels are currently whitelisted.

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -16,13 +20,15 @@ function onOpen (soc, channelIDs) {
...
-        });
+        if (!_.includes(whitelistChannels, id)) {
+            msgBroker.push({
+                "id": 1,
+                "type": "message",
+                "channel": id,
+                "text": "WhoopBot " + process.env.VERSION +
+                    " connected to WebSocket"
+            });
+        }
```
Comment (waisbrot) [inline app.js:33]: Shouldn't !includes mean that it fires if the channel is _not_ whitelisted?

File: plugins/index.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
+const github = require('./github.js');
+const httpcat = require('./httpcat.js');
+const jenkins = require('./jenkins.js');
```
Comment (waisbrot) [inline plugins/index.js]: This is better, but I was actually thinking of using https://nodejs.org/docs/latest-v5.x/api/fs.html#fs_fs_readdirsync_path to find plugins at start-time.

File: app.js (hunk 1 above)
Comment (janetchen1) [inline app.js:33]: Yes, did I invert what whitelisted means? I was thinking that whitelistedChannels would be a list of banned channels; do you think it should be a list of allowed channels?

Comment (jblanchette) [inline app.js:33]: Yeah, being on a "whitelist" means you're in / approved

**Pull request merged**